src/test/regress/expected/failure_truncate.out

```diff
@@ -0,0 +1,1094 @@
+ shardid | shardstate | shardlength | nodename  | nodeport | placementid 
+---------+------------+-------------+-----------+----------+-------------
+  102008 |          1 |           0 | localhost |    57637 |           1
+  102009 |          1 |           0 | localhost |    57640 |           2
+  102010 |          1 |           0 | localhost |    57637 |           3
+  102011 |          1 |           0 | localhost |    57640 |           4
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             2
+(1 row)
+
+SELECT count(*) FROM test_table;
+ count 
+-------
+    20
+(1 row)
+
+-- cancelling on PREPARE should be fine, everything should be rollbacked
+SELECT citus.mitmproxy('conn.onCommandComplete(command="PREPARE TRANSACTION").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+TRUNCATE test_table;
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT * FROM pg_dist_shard_placement ORDER BY 1;
+ shardid | shardstate | shardlength | nodename  | nodeport | placementid 
+---------+------------+-------------+-----------+----------+-------------
+  102008 |          1 |           0 | localhost |    57637 |           1
+  102009 |          1 |           0 | localhost |    57640 |           2
+  102010 |          1 |           0 | localhost |    57637 |           3
+  102011 |          1 |           0 | localhost |    57640 |           4
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             1
+(1 row)
+
+SELECT count(*) FROM test_table;
+ count 
+-------
+    20
+(1 row)
+
+-- killing on command complete of COMMIT PREPARE, we should see that the command succeeds
+-- and all the workers committed
+SELECT citus.mitmproxy('conn.onCommandComplete(command="COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+TRUNCATE test_table;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- we shouldn't have any prepared transactions in the workers
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             0
+(1 row)
+
+SELECT count(*) FROM test_table;
+ count 
+-------
+     0
+(1 row)
+
+INSERT INTO test_table SELECT x,x FROM generate_series(1,20) as f(x);
+-- kill as soon as the coordinator sends COMMIT
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+TRUNCATE test_table;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT recover_prepared_transactions();
```

Comment:
Minor: maybe briefly mention that this is the expected behaviour since some of the placements have been committed and some not. So, we have some pending prepared transactions.

(I think if we had client_min_messages = `WARNING`, we'd have seen the logs but then the prepared tx number changes, that's why we don't want?)